tests/integration_tests/synthetic_test_data.py

```diff
@@ -0,0 +1,71 @@
+"""Utilities for producing synthetic test data that is convergence-friendly."""
+
```

Comment:
Should we move this into `integration_tests/utils.py?`. Alternatively, we can rename this to `util_synthetic_test_data.py`. I'm personally in favor of the former

Reply:
Since this is a self-encapsulated set of library functions, I'd actually prefer to keep it in a separate file, also since `integration_tests/utils.py` is already quite large.